src/google/protobuf/util/internal/default_value_objectwriter.cc

```diff
@@ -363,8 +363,9 @@ void DefaultValueObjectWriter::Node::PopulateChildren(
     }
 
     // If oneof_index() != 0, the child field is part of a "oneof", which means
-    // the child field is optional and we shouldn't populate its default value.
-    if (field.oneof_index() != 0) continue;
+    // the child field is optional and we shouldn't populate its default
+    // primitive value.
+    if (field.oneof_index() != 0 && kind == PRIMITIVE) continue;
```

Comment:
Need to create the child node for message field so DefaultValueObjectWriter can go over the tree